tmt/steps/prepare/artifact/__init__.py

```diff
@@ -57,3 +58,18 @@ def go(
             # TODO: Not using exclude_pattern yet.
             provider.fetch_contents(guest, download_path)
         return outcome
+
+    def essential_requires(self) -> list[tmt.base.Dependency]:
+        """
+        Collect all essential requirements of the plugin.
+
+        Essential requirements of a plugin are necessary for the plugin to
+        perform its basic functionality.
+
+        :returns: a list of requirements.
+        """
```

Comment:
```suggestion
```
Please don't repeat inherited docstrings.

Reply:
Removed in next commit. Makes sense,  I only kept it because it was copied in all the other places as well. [See](https://github.com/search?q=repo%3Ateemtee%2Ftmt+%22Collect+all+essential+requirements+of+the+plugin%22&type=code).